### ISSUE TITLE
error at cbind step in interest function 

### ISSUE THREAD

**Issue by ganguvamshi** [gacatag/IntEREst#6]
Error while running interest function.

```
#preparing reference using ensembl
ref <- referencePrepare(sourceBuild="biomaRt", biomartDataset="dmelanogaster_gene_ensembl", biomart="ENSEMBL_MART_ENSEMBL", collapseExons=T)

#wrapper function to run interest for multiple bams
run_interest <- function(bamF, baseName="test", ncpu=10){
    outtsv = paste(baseName,"intRetRes.tsv", sep="_")
    outlog = paste(baseName, "log.txt", sep="_")
    interest_res <- IntEREst::interest(
        bamFileYieldSize=10000,
        junctionReadsOnly=FALSE,
        bamFile=bamF,
        isPaired=TRUE,
        isPairedDuplicate=FALSE,
        isSingleReadDuplicate=NA,
        reference=dm6_ref,
        referenceGeneNames=ref[,"collapsed_gene_id"],
        referenceIntronExon=ref[,"int_ex"],
        repeatsTableToFilter=c(),
        outFile=paste(outDir,
            outtsv, sep="/"),
        logFile=paste(outDir,
            outlog, sep="/"),
        method="IntRet",
        returnObj=FALSE,
        scaleLength= TRUE,
        sampleName=baseName,
        scaleFragment= TRUE,
        clusterNo=ncpu,
    )
    return(interest_res)
}

d10_r1_Interest <- run_interest(d10_files[1], baseName = "d10_r1")

```
InERESt: Running bamPreprocess. Detailed log info are written in:  /root/IntronRetention/IntEREst/interestFolder/d10_r1_log.txt
Log info: Running interest in Parallel mode.
InERESt: Running interestAnalyse.
InERESt:interestAnalyse: Begins ...
InERESt:interestAnalyse: Read counting ends. Running time:  2087.768  secs
InERESt: Running interestSummarise.
IntERESt:interestSummarise: Begins ...
Error in data.frame(..., check.names = FALSE) :
  arguments imply differing number of rows: 117141, 0
In addition: Warning message:
In bploop.iterate(structure(list(), class = "iterate"), ITER, FUN,  :
  first invocation of 'ITER()' returned NULL

The log file doesnt record the error.

**Comment by gacatag** [comment]
Hello.
Could you please send me the results of the following command:
`samtools view YOURBAMFILE | head -n 10`
in shell and get the first several rows of `ref` in R. You can write In your R session:
`ref [1:6,]`
or
`head(ref)`

I am basically wondering: 1. whether the chromosome names in bam file and reference match and 2. whether the reads names (IDs) for read1 and read2 are identical.

**Comment by ganguvamshi** [comment]
Hi Ali,

Thanks. 

I think the chromosome names match . 

Here is the bam snippet:

![image](https://user-images.githubusercontent.com/1865767/52403543-e943b100-2b01-11e9-9ec5-167ad08b4396.png)

And the ref snippet:

![image](https://user-images.githubusercontent.com/1865767/52403729-5f481800-2b02-11e9-95e9-81fc0d90978b.png)

**Comment by gacatag** [comment]
Paired Read names seem identical. Chromosome names also match.
just wondering; is `dm6_ref` in your code (`reference=dm6_ref`) identical to `ref` ?
If yes, it would be a bit difficult to guess what the problem is.
Is it possible for you to send me a bam file and `reference` (as R object or tsv) ? I can provide you access to a link to send me large bam file if you wish.

**Comment by ganguvamshi** [comment]
yes, sorry for the confusion, both dm6_ref and ref are same. Please share the link, I can share the reference R object and bam for your troubleshooting

**Comment by gacatag** [comment]
Cheers dear Vamshidhar.
I emailed you the links (to your gmail).

**Comment by gacatag** [comment]
Dear @ganguvamshi ,
Thank you for bringing up the issue and sending me the files. I managed to track down and correct the error.
When a single end mapped read (i.e. a read that its pair is not mapped) is analysed, if the bam file fully lacks any of these reads it would generate NULL which caused problems eventually. 

This is corrected now but I updated the changes to the develop version, [IntEREst devel](https://www.bioconductor.org/packages/devel/bioc/html/IntEREst.html) . These changes would be affected with a couple of days of delay however. 

I also pushed the changes to the develop version in GITHUB ( [master branch](https://github.com/gacatag/IntEREst) ) which is available immediately. I will make these changes to the [release version](https://www.bioconductor.org/packages/release/bioc/html/IntEREst.html) next week. I need to run tests to make sure it is stable before I publish the release.

Feel free to let me know if there were other problems.

Thank you again!
